tests/test_generators.py

```diff
@@ -0,0 +1,234 @@
+                "chars": ["4", "3", "0", "7", ".", "5", "1", "8", "2", "6", "9"],
+                "str_len_min": 11.0,
+                "str_len_max": 15.0,
+                "order": "random",
+            },
+            {
+                "generator": "categorical",
+                "name": "cat",
+                "categories": [
+                    "Seoul",
+                    "Jiangxi Sheng",
+                    "Taipei",
+                    "Oregon",
+                    "Illinois",
+                    "Henan Sheng",
+                    "Sichuan Sheng",
+                    "Hebei",
+                    "Liaoning",
+                    "Washington",
+                ],
+                "probabilities": [
+                    0.3333333333333333,
+                    0.13333333333333333,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                    0.06666666666666667,
+                ],
+                "order": "random",
+            },
+            {
+                "generator": "categorical",
+                "name": "cat",
+                "categories": ["11", "36", "OR", "IL", "41", "51", "13", "21", "WA"],
+                "probabilities": [
+                    0.35714285714285715,
+                    0.14285714285714285,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                    0.07142857142857142,
+                ],
+                "order": "random",
+            },
+            {
+                "generator": "integer",
+                "name": "int",
+                "min_value": 60661.0,
+                "max_value": 98168.0,
+                "order": "descending",
+            },
+            {
+                "generator": "float",
+                "name": "flo",
+                "min_value": 25.0392,
+                "max_value": 51.0,
+                "sig_figs": 6,
+                "order": "random",
+            },
+            {
+                "generator": "float",
+                "name": "flo",
+                "min_value": -122.9117,
+                "max_value": 127.02,
+                "sig_figs": 7,
+                "order": "random",
+            },
+            {
+                "generator": "integer",
+                "name": "int",
+                "min_value": 664.0,
+                "max_value": 9464.0,
+                "order": "random",
+            },
+        ]
+        generator.synthesize(20)
+
+        mock_generate_dataset.assert_called_once()
+        for i in range(len(expected_columns_to_generate)):
```

Comment:
testing each entry of expected with actual and then checking if the lists are equal by checking if actual is a subset of expected.